pkg/controllers/managementuser/nodesyncer/nodessyncer.go

```diff
@@ -676,6 +678,24 @@ func cleanStatus(machine *v3.Node) {
 	for _, condition := range machine.Status.InternalNodeStatus.Conditions {
 		if condition.Type == "Ready" {
 			conditions = append(conditions, condition)
+			readyCondition := v32.NodeCondition{
+				Type:               cond.Cond(condition.Type),
+				Status:             condition.Status,
+				LastTransitionTime: condition.LastTransitionTime.String(),
+				LastUpdateTime:     condition.LastHeartbeatTime.String(),
+				Reason:             condition.Reason,
+				Message:            condition.Message,
+			}
+			var exists bool
+			for i, cond := range machine.Status.Conditions {
+				if cond.Type == "Ready" {
+					exists = true
+					machine.Status.Conditions[i] = readyCondition
```

Comment:
Can we break out of loop here as we've already found `Ready` condition? Not sure if it's theoretically possible to have multiple conditions with the same type.

Reply:
done